File: Frontend/src/models/ficheDePaie.js
class FicheDePaie {
    constructor(
      employe = null,
      salaire_base = 0,
      jour_heure_travaille = 0,
      salaire = 0,
      taux_heure_sup = 0,
      heures_sup = 0,
      prix_tot_sup = 0,
      prime_transport = 0,
      prime_presence = 0,
      acompte = 0,
      impots = 0,
      apoint = 0,
      css = 0,
      cnss = 0,
      jour_ferie = 0,
      prix_jour_ferie = 0,
      prix_tot_ferie = 0,
      conge_paye = 0,
      jour_abcense = 0,
      prix_conge_paye = 0,
      prix_tot_conge = 0,
      salaire_brut = 0,
      salaire_imposable = 0,
      salaire_net = 0,
      mode_paiement = ""
    ) {
      this.employe = employe;
      this.salaire_base = salaire_base;
      this.jour_heure_travaille = jour_heure_travaille;
      this.salaire = salaire;
      this.taux_heure_sup = taux_heure_sup;
      this.heures_sup = heures_sup;
      this.prix_tot_sup = prix_tot_sup;
      this.prime_transport = prime_transport;
      this.prime_presence = prime_presence;
      this.acompte = acompte;
      this.impots = impots;
      this.apoint = apoint;
      this.css = css;
      this.cnss = cnss;
      this.jour_ferie = jour_ferie;
      this.prix_jour_ferie = prix_jour_ferie;
      this.prix_tot_ferie = prix_tot_ferie;
      this.conge_paye = conge_paye;
      this.jour_abcense = jour_abcense;
      this.prix_conge_paye = prix_conge_paye;
      this.prix_tot_conge = prix_tot_conge;
      this.salaire_brut = salaire_brut;
      this.salaire_imposable = salaire_imposable;
      this.salaire_net = salaire_net;
      this.mode_paiement = mode_paiement;
    }
  }
  
  export default FicheDePaie;

File: Backend/app/salaire/fichedepaie.py
import jinja2
from pylatex import Document
from pylatex.utils import NoEscape

data = {
    "mois": "Mars",
    "annee": "2025",
    "nom": "John Doe",
    "situationFamiliale": "Célibataire",
    "cnss": "0065113672",
    "matricule": "123456",
    "createdAt": "01/01/2020",
    "salaireContrat": "3000.00",
    "serviceNom": "Développement",
    "departementNom": "IT",
    "ribBancaire": "1234 5678 9101 1121",
    "jourTotal": "22",
    "jourFerie": "2",
    "jourConge": "1",
    "jourAbcense": "0",
    "soldeConge": "10",
    "nbJourBase": "22",
    "salaireBase": "3000.00",
    "primePresence": "200.00",
    "primeTransport": "100.00",
    "salaireBrut": "3300.00",
    "retenueCNSS": "320.00",
    "salaireImposable": "2980.00",
    "appointPlus": "150.00",
    "acompte": "0.00",
    "impoRev": "50.00",
    "appointMoins": "0.00",
    "CSS": "30.00",
    "salaireNet": "2600.00",
    "modePaiment": "Virement Bancaire"
}

template_loader = jinja2.FileSystemLoader(searchpath="./")
template_env = jinja2.Environment(loader=template_loader)
template = template_env.get_template("fichedepaie.tex")

rendered_tex = template.render(data)

doc = Document()

doc.dumps = lambda: rendered_tex

doc.generate_pdf("output", clean_tex=True, compiler="pdflatex")

print("PDF generated successfully!")

#gotta install texlive and texlive-collection-langfrench